components/merkledb/src/views/metadata.rs

```diff
@@ -294,6 +339,45 @@ impl<T: RawAccess> IndexesPool<T> {
     }
 }
 
+/// Obtains `object_hash` for an aggregated index.
+pub fn get_object_hash<T: RawAccess>(access: T, addr: ResolvedAddress) -> Hash {
+    use crate::{ObjectHash, ProofListIndex, ProofMapIndex};
+
+    let index_full_name = addr.name.as_bytes().to_vec();
```

Comment:
Is it necessary to do allocation here?

Reply:
It's used later to construct `ViewWithMetadata`.